record-and-playback/presentation/playback/presentation/0.9.0/playback.js

```diff
@@ -379,13 +380,70 @@ load_deskshare_video = function () {
    $('#video').on("play", function() {
        Popcorn('#deskshare-video').play();
    });
+
    $('#video').on("pause", function() {
        Popcorn('#deskshare-video').pause();
+
+       //here we sync medias
+       //setting currentTime on deskshare video will fire a 'canplaythrough' event after the seeking is done.
+       Popcorn("#deskshare-video").currentTime(Popcorn('#video').currentTime());
+   });
+
+   Popcorn("#deskshare-video").on("canplayall", function() {
+      //when deskshare video is ready for the first time, sync times
+      Popcorn("#deskshare-video").currentTime(Popcorn('#video').currentTime());
+   });
+
+   Popcorn("#deskshare-video").on("canplaythrough", function() {
+      console.log("deskshare-video: canplaythrough event");
+
+      if(syncing) {
+         syncing = false;
+         if(Popcorn("#video").media.readyState == 4)
+            Popcorn('#video').play();
+         else console.log("deskshare-video canplaythrough | Not resuming playback: main media source not ready");
+      }
+   });
+
+   Popcorn("#deskshare-video").on("waiting", function() {
+      //must sync deskshare video
+      console.log("deskshare-video: waiting event | CURRENT time = " + secondsToHHMMSS(Math.ceil(Popcorn("#deskshare-video").currentTime())) + " | LOADED time = " + getLoadedVideoTime("#deskshare-video"));
+      resync();
+   });
+
+   Popcorn("#video").on("waiting", function() {
+      //must sync video
+      console.log("video: waiting event | CURRENT time = " + secondsToHHMMSS(Math.ceil(Popcorn("#video").currentTime())) + " | LOADED time = " + getLoadedVideoTime("#video"));
+      resync();
    });
 
    document.dispatchEvent(new CustomEvent('media-ready', {'detail': 'deskshare'}));
 }
 
+function resync() {
+    syncing = true;
+    if(!Popcorn('#video').paused() && !Popcorn('#video').seeking())
+        Popcorn("#video").pause();
+    else console.log("resync | main video seeking or already paused");
+}
```

Comment:
Same thing here, we're only considering to enter resync after a waiting event on #deskshare-video.